src/test/ui/type/issue-103271.rs

```diff
@@ -0,0 +1,6 @@
+fn main() {
+    let length = <&[u32]>::len;
```

Comment:
I think you need to change the `len` to another function that does not interfere with the trait from the sysroot.

Reply:
Changed to `iter`